ke/compiler.py

```diff
@@ -89,6 +89,47 @@
     builtin_macros["#" + short] = builtin_macros["#" + long]
 
 
+def compile_separate(separate, expr):
+    print('\n\n\n\n#########')
+
+    if separate is None:
+        raise CompileError("Must specify a valid separator: ',', ':', '|'")
+    print(f"incoming separator of: '{separate}' - '{expr}'")
+    (a, b, is_greedy, sub) = expr
+    if sub is None:
+        return CompileError("Must specify a token")
+
+    if b is not None and a > b:
+        return CompileError("Min can not exceed Max.")
+    
+    # we are ok with max of 0, we just send an empty response.
+    if b == 0:
+        return asm.Literal("") 
+    print(f'content: {a}, {b}, {sub}')
+
+    #special case whereas we only have a single separated content.
+
+    if a == 0 and b == 1:
+        return asm.Multiple(a,b,True, sub)
```

Comment:
do we really need this special case?

Reply:
also, lets use the `is_greedy` we got from expr

Reply:
We currently do, as it helps us wrapping up properly this situation:
`AssertionError: assert 'name|' == '(?:name)?'`

Might be able to figure out a bypass for that. Also, passing in is_greedy